src/composable_kernel/composable_kernel/src/kernel_wrapper/gridwise_generic_reduction_first_call_warpwise_reduce_all_dims.cpp

```diff
@@ -177,17 +171,15 @@ extern "C" __global__ void gridwise_generic_reduce_1_prepare(int GridSize,
     }
 };
 
-template <index_t srcDims, index_t dstDims, typename toReduceDims>
+template <index_t srcDims>
 struct get_ref_desc_types
 {
     static constexpr auto ref_srcLengths = typename uniform_sequence_gen<srcDims, 8>::type{};
-    static constexpr auto ref_dstLengths = typename uniform_sequence_gen<dstDims, 1>::type{};
 
     // don't have to use accurate strides to get an expected referrence type
     static constexpr auto ref_srcDesc = make_naive_tensor_descriptor(
         make_tuple_from_seq(ref_srcLengths), make_tuple_from_seq(ref_srcLengths));
-    static constexpr auto ref_dstDesc = make_naive_tensor_descriptor(
-        make_tuple_from_seq(ref_dstLengths), make_tuple_from_seq(ref_dstLengths));
+    static constexpr auto ref_dstDesc = make_naive_tensor_descriptor(make_tuple(1), make_tuple(1));
```

Comment:
This change seems questionable to me (but I may not be familiar in the BN domain).

Reply:
This is for the reducing all dimensions cases where the output tensor has only one dimension.

Reply:
warpwise wrapper was ignored to be changed from a generic template to a more specific instantiation by previous P.R. , while changes have been made with the  threadwise/blockwise/blockwise wrapper.